librz/util/path.c

```diff
@@ -318,3 +316,34 @@ RZ_API RZ_OWN char *rz_path_realpath(RZ_NULLABLE const char *path) {
 #endif
 	return NULL;
 }
+
+/**
+ * \brief Return new RzPathPortable
+ * \return New RzPathPortable
+ */
+RZ_API RZ_OWN RzPathPortable *rz_path_portable_new(bool recursive) {
+	RzPathPortable *p = RZ_NEW0(RzPathPortable);
+	if (!p) {
+		return NULL;
+	}
+	p->prefix = NULL;
+	p->prefix_searched = false;
+	p->prefix_mutex = rz_th_lock_new(recursive);
+
+	return p;
+}
+
+/**
+ * \brief Deallocate memory associated with a RzPathPortable object
+ * \return NULL
+ */
+RZ_API RZ_OWN RzPathPortable *rz_path_portable_free(RzPathPortable *p) {
+	if (!p) {
+		return NULL;
+	}
+	RZ_FREE(p->prefix);
```

Comment:
Call free. Use RZ_FREE/RZ_FREE_CUSTOM only if you intend to reuse the variables immediately afterward.

If the structure is then freed, there is no need to waste time setting them to NULL like these macros do